addons/hashes/fnc_parseJSON.sqf

```diff
@@ -27,246 +27,524 @@ Examples:
     (end)
 
 Author:
-    BaerMitUmlaut
+    initramfs
 ---------------------------------------------------------------------------- */
 SCRIPT(parseJSON);
-params ["_json", ["_objectType", 0]];
+params [["_json", "", [""]], ["_objectType", 0]];
+
+// Enable unicode processing
+forceUnicode 0;
+
+/**
+ * Helper function returning the code used for object creation based on the "_objectType" parameter.
+ */
+private _createObject = {
+    switch (_objectType) do {
+        case false;
+        case 0: {
+            [] call CBA_fnc_createNamespace;
+        };
 
-// Wrappers for creating "objects" and setting values on them
-private ["_objectSet", "_createObject"];
+        case true;
+        case 1: {
+            [] call CBA_fnc_hashCreate;
+        };
 
-switch (_objectType) do {
-    case false;
-    case 0: {
-        _createObject = CBA_fnc_createNamespace;
-        _objectSet = {
-            params ["_obj", "_key", "_val"];
-            _obj setVariable [_key, _val];
+        case 2: {
+            createHashMap;
         };
     };
+};
 
-    case true;
-    case 1: {
-        _createObject = CBA_fnc_hashCreate;
-        _objectSet = CBA_fnc_hashSet;
-    };
+/**
+ * Helper function to set a key-value pair within an object based on the "_objectType" parameter.
+ */
+private _objectSet = {
+    params ["_obj", "_key", "_val"];
+
+    switch (_objectType) do {
+        case false;
+        case 0: {
+            _obj setVariable [_key, _val];
+        };
 
-    case 2: {
-        _createObject = { createHashMap };
-        _objectSet = {
-            params ["_obj", "_key", "_val"];
+        case true;
+        case 1: {
+            [_obj, _key, _val] call CBA_fnc_hashSet;
+        };
+
+        case 2: {
             _obj set [_key, _val];
         };
     };
 };
 
-// Handles escaped characters, except for unicode escapes (\uXXXX)
-private _unescape = {
-    params ["_char"];
-
-    switch (_char) do {
-        case """": { """" };
-        case "\": { "\" };
-        case "/": { "/" };
-        case "b": { toString [8] };
-        case "f": { toString [12] };
-        case "n": { endl };
-        case "r": { toString [13] };
-        case "t": { toString [9] };
-        default { "" };
+/**
+ * Tokenizes the input JSON string into an array of tuples of the format: (token, [row, column]).
+ *
+ * @param _jsonStr The JSON string to tokenize.
+ * @return An array of token tuples.
+ */
+private _tokenize = {
+    params [["_jsonStr", objNull, [""]]];
+
+    if (!(_jsonStr isEqualType "")) exitWith {
```

Comment:
```suggestion
    if !(_jsonStr isEqualType "") exitWith {
```

Reply:
Was not aware if statement expressions work without parenthesis (the wiki's examples use parens for trivial examples as well). Fixed.